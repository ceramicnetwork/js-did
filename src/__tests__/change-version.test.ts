import { changeVersion } from '../change-version.util'

test('from 0 to CID', () => {
  const kid =
    'did:3:kjzl6cwe1jw14acsxto0wvxwl62usq32y4ncfonzdmg5i5jorjcs2haniuabvzl?version-id=0#dkbnB3Dd1z5wByn'
  const nextVersion = 'bafyreihoahs3ge57ud5kmvs76k2dhuu46fhpilo4r7lol6fuaxh7sxk5nm'
  const result = changeVersion(kid, nextVersion)
  expect(result).toMatchSnapshot()
})

test('from CID to 0', () => {
  const kid =
    'did:3:kjzl6cwe1jw14acsxto0wvxwl62usq32y4ncfonzdmg5i5jorjcs2haniuabvzl?version-id=bafyreihoahs3ge57ud5kmvs76k2dhuu46fhpilo4r7lol6fuaxh7sxk5nm&other=foo#dkbnB3Dd1z5wByn'
  const nextVersion = '0'
  const result = changeVersion(kid, nextVersion)
  expect(result).toMatchSnapshot()
})
