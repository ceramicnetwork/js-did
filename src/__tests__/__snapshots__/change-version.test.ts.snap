// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`from 0 to CID 1`] = `"did:3:kjzl6cwe1jw14acsxto0wvxwl62usq32y4ncfonzdmg5i5jorjcs2haniuabvzl?version-id=bafyreihoahs3ge57ud5kmvs76k2dhuu46fhpilo4r7lol6fuaxh7sxk5nm#dkbnB3Dd1z5wByn"`;

exports[`from CID to 0 1`] = `"did:3:kjzl6cwe1jw14acsxto0wvxwl62usq32y4ncfonzdmg5i5jorjcs2haniuabvzl?other=foo&version-id=0#dkbnB3Dd1z5wByn"`;
